src/Moq/ExpressionExtensions.cs

```diff
@@ -421,5 +421,10 @@ public static string ToStringFixed(this Expression expression)
 		{
 			return new StringBuilder().AppendExpression(expression).ToString();
 		}
+
+		public static object[] ExtractConstructorCallArguments<T>(this Expression<Func<T>> constructorExpression)
```

Comment:
Consider renaming the parameter from `constructorExpression` &rarr; `newExpression` (here, and in the added public methods), as the latter would be more accurate: The expressions don't merely refer to a constructor; they should be `new` expressions (which in turn refer to a constructor).